UM/Qt/QtApplication.py

```diff
@@ -267,14 +268,23 @@ def recentFiles(self) -> List[QUrl]:
         return self._recent_files
 
     def _onJobFinished(self, job: Job) -> None:
-        if (not isinstance(job, ReadMeshJob) and not isinstance(job, ReadFileJob)) or not job.getResult():
+        if isinstance(job, WriteFileJob):
+            if not job.getResult() or not job.getAddToRecentFiles():
+                # For a write file job, if it failed or it doesn't need to be added to the recent files list, we do not
+                # add it.
+                return
+        elif (not isinstance(job, ReadMeshJob) and not isinstance(job, ReadFileJob)) or not job.getResult():
```

Comment:
isinstance accepts a tuple of objects, so this could be reduced to isinstance(job, (ReadMeshJob, ReadFileJob)).